### PR TITLE
Issue 8163: Correction of broken URL with proper URL in the Pravega documentation

### PR DESCRIPTION
**Pull request by amit-kumar59**
**Change log description**  
Correction of broken URL with proper URL in the Pravega documentation.

**Purpose of the change**  
 Fixes https://github.com/pravega/pravega/issues/7298

**What the code does**  
Wherever found broken URL corrected with the right URL.

**How to verify it**  
You can click on the URL it should supposed to redirect to the proper page.

### CODE REVIEW

File: documentation/src/docs/pravega-concepts.md [hunk 1]
```diff
@@ -227,7 +227,7 @@ state. Once a Checkpoint has been completed, the application can use the
 Checkpoint to reset all the Readers in the Reader Group to the known consistent
 state represented by the Checkpoint.
 
-For more details on working with Reader Groups, Please see [Reader Group Basics](basic-reader-and-writer.md#readergroup-basics).
+For more details on working with Reader Groups, Please see [Reader Group Basics](basic-reader-and-writer.md#reader-group-basics).
```
Comment (RaulGracia) [inline documentation/src/docs/pravega-concepts.md:230]: `Please`-> `please`

File: documentation/src/docs/security/securing-distributed-mode-cluster.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ limitations under the License.
 
 ## Introduction
 
-In the [distributed mode](../deployment/deployment.md#pravega-modes) of running a Pravega cluster, each service runs
+In the [distributed mode](../deployment/deployment.md#Pravega Modes) of running a Pravega cluster, each service runs
```
Comment (RaulGracia) [inline documentation/src/docs/security/securing-distributed-mode-cluster.md:29]: This link is broken, it does not render as a link.

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Thanks for improving the documentation. I have some comments that may need to be addressed.

File: documentation/src/docs/controller-service.md [hunk 3]
```diff
@@ -89,7 +89,7 @@ more than one instance of Controller service per cluster.
 
 Each Controller instance is capable of working independently and uses a
 shared persistent store as the source of truth for all state-owned and
-managed by Controller service. We use [Pravega KeyValue Tables](https://github.com/pravega/pravega/wiki/PDP-39-Key-Value-Tables) as the
+managed by Controller service. We use [Pravega KeyValue Tables](https://github.com/pravega/pravega/wiki/PDP-39-(Key-Value-Tables-Beta-1)) as the
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/controller-service.md]: Made the change.

File: documentation/src/docs/controller-service.md [hunk 4]
```diff
@@ -145,7 +145,7 @@ Transaction management can be found later in the [Transactions](#transaction-man
...
+Apart from Stream abstraction Controller is also the source of truth for the other storage primitive offered by Pravega - [Key Value Tables](https://github.com/pravega/pravega/wiki/PDP-39-(Key-Value-Tables-Beta-1)). Just like a Stream, a Key Value Table (KVT) is also distributed and paritioned using a Table Segment. A Table Segment has same properties as a Stream Segment, but with the data is formatted as keys and values and indexed on the keys. It provides APIs to perform CRUD operations on keys. Controller uses Table Segments to create a higher level abstraction which creates a distributed Key Value Table. Each Table is partitioned and the user data is distributed across different partitions using a hashing scheme. It also introduces a concept of _key family_ which is used to ensure that all keys from the same key family are always mapped to the same partition.
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/controller-service.md]: Removed.

File: documentation/src/docs/getting-started/pravega-on-kubernetes-101.md [hunk 5]
```diff
@@ -321,6 +321,6 @@ suggest you to continue exploring Pravega with the following documents:
 
 - [Deploying Pravega](../deployment/deployment.md): A more advanced guide on how to deploy Pravega on Kubernetes
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/getting-started/pravega-on-kubernetes-101.md:322]: Subtitles are corrected now.

File: documentation/src/docs/javadoc.md [hunk 6]
```diff
@@ -23,7 +23,7 @@ the need to deploy one storage service per use-case. At the moment, the Pravega
 following abstractions:
 
 ## Event Stream Client
-_[Event Stream Client](https://github.com/pravega/pravega/tree/master/client/src/main/java/io/pravega/client/stream)_
+_[Event Stream Client](pravega-concepts.md/#Streams)_
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/javadoc.md]: Made the change.

File: documentation/src/docs/javadoc.md [hunk 7]
```diff
@@ -40,13 +40,13 @@ The Batch Client allows:
 - generating a streamCut that is a bounded distance from another streamCut.
 
 ## Byte Client
-_[Byte Client](https://github.com/pravega/pravega/wiki/PDP-30-ByteStream-API)_ is a byte-oriented API providing a way 
+_[Byte Client](https://github.com/pravega/pravega/wiki/PDP-30-(ByteStream-API))_ is a byte-oriented API providing a way
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/javadoc.md]: Added.

File: documentation/src/docs/javadoc.md [hunk 8]
```diff
@@ -40,13 +40,13 @@ The Batch Client allows:
...
 ## Byte Client
-_[Byte Client](https://github.com/pravega/pravega/wiki/PDP-30-ByteStream-API)_ is a byte-oriented API providing a way 
+_[Byte Client](https://github.com/pravega/pravega/wiki/PDP-30-(ByteStream-API))_ is a byte-oriented API providing a way 
 to write data to Pravega without writing events. It presents an interface of a InputStream and an OutputStream. Data 
 written in this way is not framed or interpreted by Pravega. So there are no length headers or event boundaries. As 
 such byte offsets into the stream are meaningful and directly exposed.
 
 ## KV Tables Client
-_[KV Tables Client](https://github.com/pravega/pravega/wiki/PDP-39-Key-Value-Tables)_: Real-world analytical 
+_[KV Tables Client](https://github.com/pravega/pravega/wiki/PDP-39-(Key-Value-Tables-Beta-1))_: Real-world analytical
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/javadoc.md]: Made the change.

File: documentation/src/docs/javadoc.md [hunk 9]
```diff
@@ -60,7 +60,7 @@ maintain disparate storage systems - both KVTs and Streams share the same underl
 Pravega Segments.
 
 ## State Synchronizer Client
-_[State Synchronizer Client](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/state/StateSynchronizer.java)_:
+_[State Synchronizer Client](state-synchronizer.md)_:
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/javadoc.md]: Replaced.

File: documentation/src/docs/security/securing-distributed-mode-cluster.md (hunk 2 above)
Comment (amit-kumar59) [reply documentation/src/docs/security/securing-distributed-mode-cluster.md:29]: I have checked and made the changes as (../deployment/deployment.md#pravega-modes) 
is not pointing any ware checked in deployment.md file there is no such tag (#pravega-modes)
so replaced with #pravega-modes and this is working checked locally.

File: documentation/src/docs/getting-started/pravega-on-kubernetes-101.md [hunk 10]
```diff
@@ -321,6 +321,6 @@ suggest you to continue exploring Pravega with the following documents:
...
 and other environments.
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/getting-started/pravega-on-kubernetes-101.md:323]: Sure, I will check with the team and if required, will made the changes accordingly.

File: documentation/src/docs/javadoc.md [hunk 11]
```diff
@@ -40,13 +42,13 @@ The Batch Client allows:
 - generating a streamCut that is a bounded distance from another streamCut.
 
 ## Byte Client
-_[Byte Client](https://github.com/pravega/pravega/wiki/PDP-30-ByteStream-API)_ is a byte-oriented API providing a way 
+_[Byte Client](https://www.pravega.io/docs/latest/javadoc/clients/io/pravega/client/ByteStreamClientFactory.htmll)_ is a byte-oriented API providing a way
```
Comment (RaulGracia) [inline documentation/src/docs/javadoc.md:45]: This link is broken, there is an extra `l` at the end of the link. Please, check all the links for issues like this.

File: documentation/src/docs/javadoc.md [hunk 12]
```diff
@@ -60,7 +62,7 @@ maintain disparate storage systems - both KVTs and Streams share the same underl
 Pravega Segments.
 
 ## State Synchronizer Client
-_[State Synchronizer Client](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/state/StateSynchronizer.java)_:
+_[State Synchronizer Client](https://www.pravega.io/docs/latest/javadoc/clients/io/pravega/client/state/StateSynchronizer.html)_:
 In many situations, a distributed application may need to share information across multiple processes that may
 update it (e.g., configuration file). In this scenario, keep the data consistent under concurrent updates is key.
 The State Synchronizer Client just solves this problem: it provides a means to have state that is synchronized between
```
Comment (RaulGracia) [inline documentation/src/docs/javadoc.md:68]: The last link in this page should be pointing to https://www.pravega.io/docs/latest/javadoc/clients/index.html, but it is not working in your branch.

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments.

File: documentation/src/docs/javadoc.md (hunk 11 above)
Comment (amit-kumar59) [reply documentation/src/docs/javadoc.md]: Removed.

File: documentation/src/docs/javadoc.md (hunk 12 above)
Comment (amit-kumar59) [reply documentation/src/docs/javadoc.md:68]: Made the changes.

**Review by tkaitchuck — APPROVED** [review]

**Review by shshashwat — APPROVED** [review]

File: documentation/src/docs/clients-and-streams.md [hunk 13]
```diff
@@ -476,7 +476,7 @@ In this example, we used [`ByteStreamWriter`](https://pravega.io/docs/latest/jav
...
+Pravega provides an additional revisioned stream client as well as [Flink Connectors](https://github.com/pravega/flink-connectors) that expose Flink APIs over the event stream and batch clients. We will cover these in more detail in future articles. [`RevisionedStreamClient`](https://www.pravega.io/docs/latest/javadoc/clients/io/pravega/client/state/RevisionedStreamClient.html) uses optimistic locking to provide strong consistency across writers, ensuring all writers have full knowledge of all stream contents as they issue writes in coordination with distributed processing. Lastly, our Flink Connectors provide end-to-end exactly once across pipelines of writers and readers, seamless integration with checkpoints and savepoints, and Table API with support for both batch and stream processing. Pravega’s 0.8 release introduced a new Key Value Table client built atop segments and state synchronizers.
```
[the remark this answers is not in the document]
Comment (amit-kumar59) [reply documentation/src/docs/clients-and-streams.md]: Made the required changes as suggested.

File: test/integration/.attach_pid4288
Comment (AnkurAnand11) [inline test/integration/.attach_pid4288:1]: Can we remove this empty file ?

Comment (amit-kumar59) [reply test/integration/.attach_pid4288:1]: removed.

**Review by AnkurAnand11 — APPROVED** [review]
LGTM

**Review by ShwethaSNayak — APPROVED** [review]
LGTM

**Pull request merged**